src/raw/error.rs

```diff
@@ -0,0 +1,27 @@
+/// An error that occurs when attempting to parse raw BSON bytes.
+#[derive(Debug, PartialEq)]
+pub enum Error {
```

Comment:
As part of refactoring this into a separate submodule, I removed the `Raw` prefix from the `Error` and `Result` types defined here. This is consistent with how we define errors elsewhere in the library (e.g. `bson::de::Error` and `bson::ser::Error`), and users will need to import it with `bson::raw::{Error, Result}` anyhow.

As for RawDocument and friends, I do not think we should remove the prefix here. `raw::Error` and `raw::Result` are essentially semantically equivalent to other `Error` and `Result` types both within this library and externally, `Document` and `RawDocument` are semantically different in significant ways. Put another way, I don't think it would be confusing for readers of code using this library to gloss over what the exact Result or Error type is given that their expectations of how they work will be correct, but glossing over whether a regular Document or a RawDocument is used seems much more dangerous.